Change `combine` and `Combiner` to accept a generator

Yash-10 opened the pull request:
This PR  fixes #756 to be able to accept a `generator` instead of a `list` as argument to `Combine` and `combiner`.
### Few details
1) The `generator` input has been converted into a list in the implementation since the idea of allowing a generator is mainly intended for user convenience.
2) The relevant `tests` and other corresponding necessary files have been modified.
3) The starting letter of all the comments in the changed files have been capitalized, wherever applicable, excluding when the comment starts with an identifier, following the PEP guidlines.

@mwcraig Could you please review the PR?
Thanks!!
<hr />

- [x] For new contributors: Did you add yourself to the "Authors.rst" file?

For new functionality:

- [x] Did you add an entry to the "Changes.rst" file?
- [ ] Did you include a meaningful docstring with Parameters, Returns and Examples? [**Modified existing docstrings**]
- [x] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [ ] Did you include tests for the new functionality? [**Modified existing tests**]
- [x] Does this PR add, rename, move or remove any existing functions or parameters?

mwcraig commented:
@Yash-10 -- it is going to take me a day or two to review this one. `combine` and `Combiner` should still be able to take a list, and I had thought this would be a fairly small change. Hopefully I'll get a look at it on Saturday, otherwise Monday.

In the meantime, can you please add yourself to the contributor list? 😃

Yash-10 commented:
Hello @mwcraig, thank you for your reply and sorry for my delayed response. As you have suggested, I have also added myself to the contributing list.

Thanks!

mwcraig commented:
I had been thinking of this as a much more limited change, basically two things:

+ add a new test that calls `Combiner` with a generator instead of a list, and that *fails* with the code currently on the main branch. I like writing tests like that first when I get a bug report to make sure that the that the test detects the bug. 
+ Adding [this line](https://github.com/astropy/ccdproc/blob/a6f90de460df3dd6f45a4b605fe482523bfe8de9/ccdproc/combiner.py#L75) to `Combiner`. 

I *think* those two changes should be all that is necessary to fix the issue. It might be easier to open a new PR with just those changes to see if it works, though you could try undoing many of the changes here and keep working on it here.

Yash-10 commented:
Hello @mwcraig, I have made the necessary changes on this same branch. I changed the name `ccd_list` to `ccd_iter`, added a test where `Combiner` accepts a generator and made some comments consistent there.

I realized that I had made changed to many files earlier because I had also changed `combine` to accept a generator which required other files to also be changed in order for the tests to pass. I think that `combine` should also be given the flexibility to accept a generator, if yes, then according to me it would require slightly manipulating some other tests as done earlier.

Please let me know what do you think about this.

Thanks!

mwcraig commented:
Thanks, @Yash-10, this is excellent! I will try to get a release out early next week. Ideally that will also include #741 but if I don't get that wrapped up I think a new release is warranted by this and some other bug fixes. One last fix -- can you please add an entry in `CHANGES.rst` about this in the section for release `2.1.1` under `Bug fixes`?

Yash-10 commented:
Thank you @mwcraig, done!

mwcraig commented:
Wonderful, thanks!
